base/linalg/dense.jl

```diff
@@ -71,7 +71,7 @@ vecnorm2{T<:BlasFloat}(x::Union{Array{T},StridedVector{T}}) =
 function triu!(M::AbstractMatrix, k::Integer)
     m, n = size(M)
     if (k > 0 && k > n) || (k < 0 && -k > m)
-        throw(BoundsError())
+        throw(BoundsError("Requested diagonal, $k, out of bounds in matrix of size ($m,$n)"))
```

Comment:
That can cause problems, if there is any code with try/catch expecting the `BoundsError`
(had to deal with that when I was dealing with trying to improve the messages in `AbstractArray`)